include/sisl/fds/buffer.hpp

```diff
@@ -74,6 +74,19 @@ struct sg_iterator {
         return ret_iovs;
     }
 
+    void move_offset(uint32_t const size) {
+        uint64_t remain_size = size;
+
+        for (; (remain_size > 0) && (m_cur_index < m_input_iovs.size()); ++m_cur_index, m_cur_offset = 0) {
+            const auto& inp_iov = m_input_iovs[m_cur_index];
```

Comment:
I see you use the preferred `auto const&` down below. Maybe consistency and fix this?